### PR TITLE
Connect wallet v2 various improvements

### PR DESCRIPTION
**Pull request by MananTank**
## react-core
- Stop forcing network switch when connecting wallet
- add autoSwitch option to force network switch in ThirdwebProvider
- remove deprecated and unused `chainRpc` prop from `ThirdwebProvider` 
- remove unused and confusing state 'activeChain' in wallet-provider
- Re export types from /sdk in /react-core

## wallets
- Fix wrong RPC urls and explorer URLs in wallets package
- make switchChain public in wallet connect v1 and remove walletName check

## react
- Add "Confirm in Wallet" UI in Web3Button, Network Switcher

## Other
- Fix hotlink DX

### CODE REVIEW

File: packages/react-core/src/core/providers/thirdweb-provider.tsx [hunk 1]
```diff
@@ -117,33 +114,22 @@ export const ThirdwebProviderCore = <
...
     if (typeof props.activeChain === "string") {
-      return (
-        supportedChains.find((chain) => chain.slug === props.activeChain) ||
-        supportedChains[0]
-      );
+      return supportedChains.find((chain) => chain.slug === props.activeChain);
     }
 
-    return props.activeChain || supportedChains[0] || defaultChains[0];
+    return props.activeChain;
```
Comment (iketw) [inline packages/react-core/src/core/providers/thirdweb-provider.tsx:132]: With this change we no longer have an `activeChain` in our SDK by default. Not sure about the downstream consequences here. cc: @jnsdls

File: packages/react-core/src/index.ts [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+
+// export types from thirdweb-dev/sdk
+export type {
+  NFTMetadataInput,
+  NFT,
+  NFTMetadata,
+  NFTMetadataOrUri,
+} from "@thirdweb-dev/sdk";
+export type { CurrencyValue, TokenMetadata } from "@thirdweb-dev/sdk";
+export type { QueryAllParams } from "@thirdweb-dev/sdk";
```
Comment (iketw) [inline packages/react-core/src/index.ts:12]: There are other types we may be missing here and that users may need. We may need to re-export more?

Some of these types: `NFTDrop`, `Token`, `SmartContract`. In SDK we do `export * from "./types";`, so there are a bunch of types also there

File: packages/react-native/src/evm/wallets/connectors/coinbase-wallet/index.ts [hunk 3]
```diff
@@ -226,7 +227,7 @@ export class CoinbaseWalletConnector extends Connector<
                 chainId: id,
                 chainName: chain.name,
                 nativeCurrency: chain.nativeCurrency,
-                rpcUrls: [chain.rpc[0] ?? ""],
+                rpcUrls: [getChainRPC(chain)],
```
Comment (iketw) [inline packages/react-native/src/evm/wallets/connectors/coinbase-wallet/index.ts:230]: Nice catch!

File: packages/wallets/src/evm/connectors/wallet-connect-v1/index.ts [hunk 4]
```diff
@@ -251,32 +247,29 @@ export class WalletConnectV1Connector extends Connector<
           `Chain ${chainId} is not added in the list of supported chains`,
         );
       }
+      console.log({ chain });
```
Comment (iketw) [inline packages/wallets/src/evm/connectors/wallet-connect-v1/index.ts:250]: Can remove this log

File: packages/react-core/src/core/providers/thirdweb-provider.tsx (hunk 1 above)
Comment (MananTank) [reply packages/react-core/src/core/providers/thirdweb-provider.tsx:132]: `WrappedThirdwebSDKProvider` handles the case where activeChain is `undefined` 

image

**Pull request merged**